pimd/pim_pim.c

```diff
@@ -124,6 +131,15 @@ void pim_sock_delete(struct interface *ifp, const char *delete_message)
 	pim_neighbor_delete_all(ifp, delete_message);
 
 	sock_close(ifp);
+
+	if (pim_ifp->gm_enable) {
+#if PIM_IPV == 4
```

Comment:
we can remove if else checks here by calling pim_gm_if_reset directly as mentioned above.